### PR TITLE
[SCHEMA] Add datatypes field to datatype rule files

### PR DESCRIPTION
**Pull request by tsalo**
Closes None.

Changes proposed:
- Add a datatypes field to the datatype rule files, so that the folder structure in the rules folder can be for readability, rather than something that implies the schema's structure.

What we won't tackle in this PR, but which is important, is that the schema code needs to be updated to use the new `datatypes` field _instead of_ the directory/filename structure that is currently used to organize the schema.

### CODE REVIEW

File: src/schema/rules/datatypes/meg.yaml [hunk 1]
```diff
@@ -60,8 +66,10 @@
 - suffixes:
   - headshape
   extensions:
-  - .*
+  - "*"
```
Comment (effigies) [inline src/schema/rules/datatypes/meg.yaml:69]: I think this was a merge conflict resolution? Seems out of scope.

```suggestion
  - .*
```

**Comment by effigies** [comment]
Plan on merging this with or without schemacode changes to make the field available to tooling. Note that merging as is will mean that we can't take advantage of the reorganization potential until there are schemacode changes.

File: src/schema/rules/datatypes/meg.yaml (hunk 1 above)
Comment (tsalo) [reply src/schema/rules/datatypes/meg.yaml:69]: Ah, yes. I think I just merged it badly.
